csv_to_json_large_convos_to_single_utterance.py

```diff
@@ -0,0 +1,259 @@
+"""
+python csv_to_json_combine_large_convos.py
+
+"""
+# *********************************************************************************************************************
+import click
+import csv
+import json
+import time
+import datetime
+import os
+
+@click.command()
+@click.option('-f', '--filename', type=str, required=True, help='Input File Path')
+@click.option('-l', '--limit', type=int, required=True, help='Conversation Split Limit Number (ex 6000)')
+@click.option('-u', '--utterance_col', type=str, required=True,
+              help='Column name containing utterances')
+@click.option('-c', '--convo_id_col', type=str, required=True, default='',
+              help='Column name containing convo id')
+@click.option('-t', '--created_at_col', type=str, required=False, default='',
+              help='If there is a created date for utterance otherwise defaults to now')
+@click.option('-x', '--unix_date', is_flag=True, type=bool, required=False, default=False,
+              help='Boolean to see if created_at column is in unix epoch format')
+@click.option('-r', '--role_col', type=str, required=False, default='',
+              help='Column name containing the role')
+@click.option('-p', '--role_mapper', type=str, required=False, default='',
+              help='If role column then role mapper in format "source_client:client,source_expert:expert,*:expert"')
+def main(filename: str, limit: int, utterance_col: str,
+         convo_id_col: str, created_at_col: str, unix_date: bool, role_col: str,
+         role_mapper: str):
+
+    file_name = filename
+    #Check if filename is a path
+    if "/" in file_name:
+        file_name = filename.split("/")[-1]
+
+    #Remove any .
+    if "." in file_name:
+        file_name_split = file_name.split(".")[0]
+    else:
+        file_name_split = file_name
+
+    json_file_name = file_name_split + "-hf-export.json"
+
+    convo_dict = combine_conversations(filename,utterance_col,convo_id_col,created_at_col,unix_date,role_col,role_mapper)
+    export_to_hf_json_limit(convo_dict,json_file_name,limit=limit)
+
+
+
+def combine_conversations(filename: str,utterance_col,convo_id_col,created_at_col,unix_date,role_col,role_mapper):
```

Comment:
provide the data type of every function variables and also the data type of the values returned. If no values are returned by the function, then provide None